golfscript.rb

```diff
@@ -420,10 +332,8 @@ def compile(tokens=scan(/[a-zA-Z_][a-zA-Z0-9_]*|'(?:\\.|[^'])*'?|"(?:\\.|[^"])*"
 	end
 end
 def gpop
-	($lb.size-1).downto(0){|i|
-		break if $lb[i]<$stack.size
-		$lb[i]-=1
-	}
+	i=$lb.size
+	$lb[i] -= 1 while i>0 && $lb[i-=1] >= $stack.size
 	$stack.pop
```

Comment:
`gpop` is called super often, so I tried rewriting it a couple of ways to squeeze more performance out (honestly mostly for fun at this point). It seems this is a bit faster than a block.